File: typespec/hub/applications.go
package hub

import (
	"time"

	"github.com/psankar/vetchi/typespec/common"
)

type MyApplicationsRequest struct {
	State         common.ApplicationState `json:"state"          validate:"omitempty,validate_application_state"`
	PaginationKey *string                 `json:"pagination_key" validate:"omitempty"`
	Limit         int64                   `json:"limit"          validate:"required,min=0,max=40"`
}

type HubApplication struct {
	ApplicationID  string                  `json:"application_id"`
	State          common.ApplicationState `json:"state"`
	OpeningID      string                  `json:"opening_id"`
	OpeningTitle   string                  `json:"opening_title"`
	EmployerName   string                  `json:"employer_name"`
	EmployerDomain string                  `json:"employer_domain"`
	CreatedAt      time.Time               `json:"created_at"`
}

type WithdrawApplicationRequest struct {
	ApplicationID string `json:"application_id" validate:"required"`
}

File: typespec/hub/applications.ts
import { ApplicationState } from '../common/applications';

export interface MyApplicationsRequest {
    state?: ApplicationState;
    pagination_key?: string;
    limit: number;
}

export interface HubApplication {
    application_id: string;
    state: ApplicationState;
    opening_id: string;
    opening_title: string;
    employer_name: string;
    employer_domain: string;
    created_at: Date;
}

export interface WithdrawApplicationRequest {
    application_id: string;
} 